codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/transport/WebSocketTransport.java

```diff
@@ -0,0 +1,230 @@
+import io.netty.handler.codec.sockjs.handler.SessionHandler.Event;
+import io.netty.handler.codec.sockjs.handler.SockJsHandler;
+import io.netty.handler.codec.sockjs.util.JsonConverter;
+import io.netty.util.AttributeKey;
+import io.netty.util.ReferenceCountUtil;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * WebSocketTransport is responsible for the WebSocket handshake and also for receiving WebSocket frames.
+ */
+public class WebSocketTransport extends ChannelInboundHandlerAdapter {
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(WebSocketTransport.class);
+    private static final AttributeKey<HttpRequest> REQUEST_KEY = AttributeKey.valueOf("request.key");
+    private final SockJsConfig config;
+    private WebSocketServerHandshaker handshaker;
+
+    public WebSocketTransport(final SockJsConfig config) {
+        checkNotNull(config, "config");
+        this.config = config;
+    }
+
+    @Override
+    public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
+        if (msg instanceof FullHttpRequest) {
+            handleHttpRequest(ctx, (FullHttpRequest) msg);
+        } else if (msg instanceof WebSocketFrame) {
+            handleWebSocketFrame(ctx, (WebSocketFrame) msg);
+        }
+    }
+
+    private static boolean checkRequestHeaders(final ChannelHandlerContext ctx, final HttpRequest req) {
+        if (req.method() != GET) {
+            logger.debug("Request was not of type GET, was {}", req.method());
+            ctx.writeAndFlush(responseFor(req)
+                    .methodNotAllowed()
+                    .header(CONTENT_LENGTH, 0)
+                    .allow(GET)
+                    .buildResponse()).addListener(ChannelFutureListener.CLOSE);
+            return false;
+        }
+
+        final String upgradeHeader = req.headers().get(HttpHeaderNames.UPGRADE);
+        if (upgradeHeader == null || !"websocket".equals(upgradeHeader.toLowerCase())) {
+            logger.debug("Upgrade header was not 'websocket' was: {}", upgradeHeader);
+            ctx.writeAndFlush(responseFor(req)
+                    .badRequest()
+                    .content("Can \"Upgrade\" only to \"WebSocket\".")
+                    .contentType(HttpResponseBuilder.CONTENT_TYPE_PLAIN)
+                    .buildFullResponse()).addListener(ChannelFutureListener.CLOSE);
+            return false;
+        }
+
+        String connectHeader = req.headers().get(CONNECTION);
+        if (connectHeader != null && "keep-alive, upgrade".equals(connectHeader.toLowerCase())) {
+            logger.debug("Connection header was not 'keep-alive, upgrade' was: {}", connectHeader);
+            req.headers().set(CONNECTION, UPGRADE);
+            connectHeader = UPGRADE.toString();
+        }
+        if (connectHeader == null || !"upgrade".equals(connectHeader.toLowerCase())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1157)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) When doing a String.toLowerCase()/toUpperCase() call, use a Locale [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseLocaleWithCaseConversions)